### DISCUSSION TITLE
Embedding videos versus adblockers

### DISCUSSION THREAD

**Discussion by eawmaas** [xenocrat/chyrp-lite#226] [General]
More and more people use browsers with build-in adblockers or browser-extensions for blocking ads.

I just noted that embedded videos with the Chyrp Lite Shorttag `<!-- video-url -->` are not only being blocked but cause 404-errors for every ad that's on it because the ad-blocker makes the browser refuse these requests, this causes 4 failed pageloads per video, I had two videos and I noticed in a request - tracking script that I wrote and use that 8 x googleads (doubleclick etc.) were blocked.

My blog started loading slowly and I tracked down the reason: with every pageload 8 errorpages are being requested as well which causes 9 pageviews on every load instead of just 1.

Next to that 4 textfiles named f.txt were blocked, I downloaded one and looking at the content I first thought my blog was hacked, this is the content:
```
)]}'

{"id":"ANyPxKrjZcMaDKFMOtTKWUCjtP9Muwx3lVVupPOFmbPQZ6HVPoxPXg0MPMHMfdR8bLT2DvCx5VED","type":0}
```
The first four characters set all my alarm-lights to red...
But it was no hacker, it was Youtube itself, when I took down the embeds these files were gone too.

How to embed a video in another way I do not know, I will take a screenshot and place an image which I link to the video, that's the safest solution.

One more thing: remember that because you are loading 3rd party-trackers by embedding these Youtube-videos you also need a user-consent script. I just dumped all external trackers ánd the consent-script two weeks ago and I am not going to put one back up just because I have added some videos...

**Comment by xenocrat** [comment]
Thanks for sharing your experience. Yes, unfortunately the embeds offered by many platforms are adtech/privacy nightmares; a good adblocker or privacy extension (e.g. Privacy Badger) can limit the intrusion, but it's a shame your visitors need to be on guard.

I think a screenshot that links to the video is a good simple solution if you don't want to host the content directly.

**Comment by eawmaas** [comment]
Yes I am doing the screenshot-thing already and it's perfect, should have done it from the start because you can choose your favorite screen from the movie as a sort of preview. I am happy, visitors happy, Youtube has to wait for their money to come from within their own domain, not mine anymore ;-) ánd you save on bandwith (handy for if you have limitations on that within your hosting-account).

**Comment by eawmaas** [comment]
Wait, something probably changed: two days ago this worked: nest an image in a <a-tag but I just noticed it doesn't anymore, I use <figure and <picture so I took those out and used only <img but still it's not going to Youtube but it shows the picture in a separate screen.
Maybe it has something to do with the new version, I am using Topaz btw.

But after a while I found out that the lightbox-module is causing the blog to show the image instead of following the URL, so if anyone has trouble with this, disable lightbox.

**Comment by xenocrat** [comment]
You can suppress that behaviour without disabling the lightbox module by adding `class="suppress_lightbox"` to the image:

    <img src="my_awesome_image.jpg" class="suppress_lightbox">

**Comment by eawmaas** [comment]
Ah okay, thanks!!! Didn't know that...
